orca-sql/src/main/kotlin/com/netflix/spinnaker/orca/sql/PipelineRefTriggerDeserializerSupplier.kt

```diff
@@ -9,42 +9,70 @@ package com.netflix.spinnaker.orca.sql
 
 import com.fasterxml.jackson.core.JsonParser
 import com.fasterxml.jackson.databind.JsonNode
-import com.netflix.spinnaker.kork.exceptions.SystemException
 import com.netflix.spinnaker.orca.api.pipeline.models.Trigger
-import com.netflix.spinnaker.orca.sql.pipeline.persistence.PipelineRefTrigger
 import com.netflix.spinnaker.orca.pipeline.model.support.CustomTriggerDeserializerSupplier
-import com.netflix.spinnaker.orca.pipeline.model.support.listValue
 import com.netflix.spinnaker.orca.pipeline.model.support.mapValue
+import com.netflix.spinnaker.orca.pipeline.model.support.listValue
+import com.netflix.spinnaker.orca.sql.pipeline.persistence.PipelineRefTrigger
 
 class PipelineRefTriggerDeserializerSupplier : CustomTriggerDeserializerSupplier {
-  override val predicate: (node: JsonNode) -> Boolean
+
+  override val rule: CustomTriggerDeserializerSupplier.OTHER_FIELD_RULE
+    get() = CustomTriggerDeserializerSupplier.OTHER_FIELD_RULE.EMPTY
+
+  override val predicateByNode: (node: JsonNode) -> Boolean
     get() = { node ->
-      node.isPipelineRefTrigger()
+      node.looksLikePipeline() || node.isPipelineRefTrigger()
+    }
+
+  override val predicateByTrigger: (trigger: Trigger) -> Boolean
+    get() = {
+      it.type == "pipelineRef"
     }
 
   override val deserializer: (node: JsonNode, parser: JsonParser) -> Trigger
     get() = { node, parser ->
+
       when {
-        node.isPipelineRefTrigger() -> {
+        node.looksLikePipeline() -> {
           with(node) {
             PipelineRefTrigger(
-              correlationId = get("correlationId")?.textValue(),
-              user = get("user")?.textValue(),
+              correlationId = get("correlationId").textValue(),
+              user = get("user").textValue(),
               parameters = get("parameters")?.mapValue(parser) ?: mutableMapOf(),
               artifacts = get("artifacts")?.listValue(parser) ?: mutableListOf(),
               notifications = get("notifications")?.listValue(parser) ?: mutableListOf(),
-              isRebake = get("rebake")?.booleanValue() == true,
-              isDryRun = get("dryRun")?.booleanValue() == true,
-              isStrategy = get("strategy")?.booleanValue() == true,
-              parentExecutionId = get("parentExecutionId").textValue(),
-              parentPipelineStageId = get("parentPipelineStageId")?.textValue()
+              isRebake = get("rebake")?.booleanValue() == false,
+              isDryRun = get("dryRun")?.booleanValue() == false,
+              isStrategy = get("strategy")?.booleanValue() == false,
+              parentExecutionId = get("parentExecution").get("id").textValue(),
+              parentPipelineStageId = get("parentPipelineStageId").textValue()
+            )
+          }
+        }
+        else -> {
+          with(node) {
```

Comment:
Why do we have an else that does the same thing?

Reply:
I must to return the trigger always. If I do not return a trigger object the Trigger will be deserializer as a Default Trigger and that is wrong.
My code is ugly here. I might the same object and conditionally populate the parentExecutionRef.